src/main/java/com/google/cloud/tools/appengine/api/auth/Auth.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.appengine.api.auth;
+
+import java.nio.file.Path;
+
+public interface Auth {
+  void login(String user);
+
+  void login();
+
+  void activateServiceAccount(Path jsonFile);
```

Comment:
I'm wondering whether we need this.  If we have the path to the json file can't we just pass it as a flag to commands like we do in IDEs?

Reply:
I think we might end up doing that.